classes/AssociationManager.php

```diff
@@ -50,32 +50,41 @@ public function getRelationshipTypes(){
-        $relationshipStr = (array_key_exists('relationship', $associationArr) && $associationArr['relationship'] !== 'any') ? ("='" . $relationshipType . "'") : ' IS NOT NULL';
+        $relationshipType = (array_key_exists($relationshipAccessor, $associationArr) && $associationArr[$relationshipAccessor] !== 'any') ? $associationArr[$relationshipAccessor] : 'IS NOT NULL';
+        $relationshipStr = (array_key_exists($relationshipAccessor, $associationArr) && $associationArr[$relationshipAccessor] !== 'any') ? ("='" . $relationshipType . "'") : ' IS NOT NULL';
 
 		$forwardSql = "SELECT oa.occid FROM omoccurrences o INNER JOIN omoccurassociations oa ON o.occid = oa.occidAssociate  LEFT JOIN omoccurdeterminations od ON oa.occid = od.occid " . $familyJoinStr . " WHERE oa.relationship " . $relationshipStr . " ";
 		$forwardSql .= $this->getAssociatedTaxonWhereFrag($associationArr);
 
         // "Reverse" association
-        $reverseAssociationType = (array_key_exists('relationship', $associationArr) && $associationArr['relationship'] !== 'any') ? $this->getInverseRelationshipOf($relationshipType) : 'IS NOT NULL';
-        $reverseRelationshipStr = (array_key_exists('relationship', $associationArr) && $associationArr['relationship'] !== 'any') ? ("='" . $reverseAssociationType . "'") : ' IS NOT NULL';
+        $reverseAssociationType = (array_key_exists($relationshipAccessor, $associationArr) && $associationArr[$relationshipAccessor] !== 'any') ? $this->getInverseRelationshipOf($relationshipType) : 'IS NOT NULL';
+        $reverseRelationshipStr = (array_key_exists($relationshipAccessor, $associationArr) && $associationArr[$relationshipAccessor] !== 'any') ? ("='" . $reverseAssociationType . "'") : ' IS NOT NULL';
 
 
 		$reverseSql = "SELECT oa.occidAssociate FROM omoccurrences o INNER JOIN omoccurassociations oa ON o.occid = oa.occid LEFT JOIN omoccurdeterminations od ON oa.occid = od.occid " . $familyJoinStr . " WHERE oa.relationship " . $reverseRelationshipStr . " ";
 
         $reverseSql .= $this->getAssociatedTaxonWhereFrag($associationArr);
 
-        $sql .= "AND (o.occid IN (SELECT occid FROM ( " . $forwardSql . " UNION " . $reverseSql . " ) AS occids)";
+		// External associations
+		// $externalAndObservationalSql = "SELECT oa.occid FROM omoccurrences o INNER JOIN omoccurassociations oa ON o.occid = oa.occid  LEFT JOIN omoccurdeterminations od ON oa.occid = od.occid " . $familyJoinStr . " WHERE (oa.associationType='observational' OR oa.associationType='externalOccurrence') AND oa.relationship " . $relationshipStr . " ";
+		$externalAndObservationalSql = "SELECT oa.occid FROM omoccurrences o INNER JOIN omoccurassociations oa ON o.occid = oa.occid  LEFT JOIN omoccurdeterminations od ON oa.occid = od.occid " . $familyJoinStr . " WHERE (oa.associationType='observational' OR oa.associationType='externalOccurrence') AND oa.relationship " . $relationshipStr . " ";
```

Comment:
This should be the only difference between 1924-export-associations-as-resource-relationship and 1924-pt2-external-and-observational-occurrences.